### ISSUE TITLE
Crash on iOS

### ISSUE THREAD

**Issue by Wackymax** [ekasetiawans/flutter_background_service#214]
Hi,

Thank you for the lib! Works brilliantly on Android. I don't need the background functionality on iOS though and it keeps crashing on my production release. Can you provide a way to choose between using the functionality for iOS or not? And do you think you will be able to fix this bug?

<img width="911" alt="image" src="https://user-images.githubusercontent.com/11558269/191112601-ea7c6164-446a-4278-ac1f-4982169c03d2.png">

**Comment by ekasetiawans** [comment]
could you provide a sample project with reproducible issue?

**Comment by Wackymax** [comment]
I can in a few hours. But you can reproduce it by not calling the initialize method or any other functionality on this lib if the platform is ios. It's like it wants to call the foreground and background methods without being instructed to do so. I'd say the library should only bootstrap if it's been initialised explicitly.On 20 Sep 2022, at 02:21, Eka Setiawan Saputra ***@***.***> wrote:﻿
could you provide a sample project with reproducible issue?

**Comment by Wackymax** [comment]
This is the piece of code that breaks. Because configure was never called the background call handle is not set. For my use case the background app refresh should not even be registered because I am using geolocator to keep the app alive on iOS for my activity tracking app.

```
        if (call.method == "getBackgroundHandler") {
            let defaults = UserDefaults.standard
            let callbackHandle = defaults.object(forKey: "background_callback_handle") as! Int64
            result(callbackHandle)
            return
        }
```

**Comment by Wackymax** [comment]
To reproduce, comment out `await initializeService();` in the example project and run the app on a clean iOS simulator. Close the app and wait for the system to trigger the background refresh logic

**Comment by Wackymax** [comment]
I used flutter_foreground_task before moving to this lib, prefer this one because of the bidirectional communication support. Maybe there are some hints there on how to only register the background fetch if/when it's needed instead of on plugin registration?

**Comment by ekasetiawans** [comment]
> This is the piece of code that breaks. Because configure was never called the background call handle is not set. For my use case the background app refresh should not even be registered because I am using geolocator to keep the app alive on iOS for my activity tracking app.
> 
> ```
>         if (call.method == "getBackgroundHandler") {
>             let defaults = UserDefaults.standard
>             let callbackHandle = defaults.object(forKey: "background_callback_handle") as! Int64
>             result(callbackHandle)
>             return
>         }
> ```

I can confirm that was a bug, I think we should handle null safety there.

**Comment by ekasetiawans** [comment]
I have published 2.3.9 to fix this issue, please let me know if your problem was resolved. Thanks.

**Comment by Wackymax** [comment]
Looks like it's fixed now. ThanksOn 21 Sep 2022, at 03:36, Eka Setiawan Saputra ***@***.***> wrote:﻿
I have published 2.3.9 to fix this issue, please let me know if your problem was resolved. Thanks.
